helix-loader/src/session.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{command_histfile, search_histfile};
+use std::{
+    fs::{File, OpenOptions},
+    io::{self, BufRead, BufReader, Write},
+    path::PathBuf,
+};
+
+pub fn push_history(register: char, line: &str) {
+    let filepath = match register {
+        ':' => command_histfile(),
+        '/' => search_histfile(),
+        _ => return,
+    };
+
+    let mut file = OpenOptions::new()
+        .append(true)
+        .create(true)
+        .open(filepath)
+        // TODO: do something about this unwrap
+        .unwrap();
+
+    // TODO: do something about this unwrap
+    writeln!(file, "{}", line).unwrap();
```

Comment:
Register contents will probably need to be encoded somehow, for example base64. You can copy line endings into the registers which will break reading below

Reply:
Good point. You ok with bincode? Since that's what I'm using anyway for files and splits

Reply:
Yep I think bincode should be ok for this too